bindings/python/capstone/__init__.py

```diff
@@ -1221,6 +1222,32 @@ def disasm(self, code, offset, count=0):
             return
             yield
 
+    # This function matches the cs_disasm_iter implementation which
+    # *should* be much faster via the C API due to pre-allocating
+    # memory (https://www.capstone-engine.org/iteration.html).
+    # Note: It is unclear whether this function via the Python
+    # binding provides the same speedup like cs_disasm_lite.
+    def disasm_iter(self, code, offset):
+        if self._diet:
+            # Diet engine cannot provide @mnemonic & @op_str
+            raise CsError(CS_ERR_DIET)
+        insn = _cs_insn()
+        size = ctypes.c_size_t(len(code))
+
+        # Pass a bytearray by reference
+        view = memoryview(code)
+        code = ctypes.pointer(ctypes.c_char.from_buffer_copy(view))
+        if view.readonly:
+            code = (ctypes.c_char * len(view)).from_buffer_copy(view)
+        else: 
+            code = ctypes.pointer(ctypes.c_char.from_buffer(view))
```

Comment:
Slightly different here than the disasm_lite implementation.
I think it's the same effect of "copy if I can't write"